lib/kuddle.ex

```diff
@@ -14,35 +14,90 @@ defmodule Kuddle do
   @doc """
   Decode a KDL document into kuddle nodes
 
-  Usage:
+  ## Usage
 
-      [%Node{name: "node"}] = Kuddle.decode("node")
+      Kuddle.decode(blob)
+      {:ok, [%Kuddle.Node{name: "node"}]} = Kuddle.decode("node")
+
+  ## Examples
+
+      iex> Kuddle.decode("node { node2 1; }")
+      {:ok, [%Kuddle.Node{name: "node", children: [%Kuddle.Node{name: "node2", attributes: [%Kuddle.Value{type: :integer, value: 1, format: :dec}]}]}], []}
 
   """
   @spec decode(String.t()) ::
           {:ok, document(), rest::String.t()}
```

Comment:
i think you missed a `rest::String.t()` > `rest::tokens()` here :)

Reply:
And fixed